Add documentation and manifests for adding new org members

hasheddan opened the pull request:
Adds documentation and an issue template for adding new members to the Crossplane org. Also introduces `Membership` manifests that will be reconciled by `provider-github`. You can see this in action on recent [CrossHack stream](https://youtu.be/osrUo-7Twz0) -- note that this is not enabled yet, this is just adding the manifests so we can start adding new members to the org via this process (+ the current manual step) while the org cluster is being set up.

Also note that the manifests contain only 24/40 members because some members have chosen to mark their membership as private. This is fine from a reconciliation perspective because provider-github will have credentials to view the private members, but out of an abundance of caution I didn't want to reflect those private members in this public repository until we can either verify that it is acceptable to do so and / or have them flip their membership to public.

Fixes #1

negz commented on config/members-crossplane.yaml:
```diff
@@ -0,0 +1,263 @@
+apiVersion: organizations.github.crossplane.io/v1alpha1
+kind: Membership
+metadata:
+  name: Chrisupbound-crossplane
```
I think these names will need to be lowercase.

hasheddan replied:
Good catch :)

negz commented on config/members-crossplane.yaml:
```diff
@@ -0,0 +1,263 @@
+apiVersion: organizations.github.crossplane.io/v1alpha1
+kind: Membership
+metadata:
+  name: bassam-crossplane
```
Would it make sense for these to read least-to-most-specific - e.g. `crossplane-bassam`? Might also be handy to label them by org so we can easily `kubectl get membership -l org=crossplane`.

hasheddan replied:
Great idea! I imagine as provider-github continues to mature we will have some XRDs that make these a little less unwieldy by doing things such as patching labels with org name, etc. -- but manually doing for now is a good idea 👍

negz reviewed (commented):
Awesome work @hasheddan! Really glad to see us working to help more folks take on more formal roles in our community.

bassam reviewed (approved):
Fantastic work Dan! Really glad to see this

jbw976 commented on .github/ISSUE_TEMPLATE/new-member.md:
```diff
@@ -0,0 +1,25 @@
+- [ ] I have [enabled 2FA](https://github.com/settings/security) on my GitHub account
+- [ ] I have joined the [Crossplane slack channel](https://slack.crossplane.io/)
+- [ ] I am actively contributing to the Crossplane project
+- [ ] I have two sponsors that are current Crossplane Org members
```
let's make sure these requirements are reflected in the governance as well: https://github.com/crossplane/crossplane/blob/master/GOVERNANCE.md#organization-members

can you make a corresponding update there? you could also consider updating that section simply to point/link to the resources in this repo, but it may be preferrable to have everything explicitly spelled out within the governance.

jbw976 commented on processes/new-member.md:
```diff
@@ -0,0 +1,19 @@
+member to the
+[config/members-crossplane.yaml](../config/members-crossplane.yaml) file.
+
+The pull request must be approved and merged by a member of the [Crossplane
```
let's make sure we have branch protection rules that enforce the ability to merge PRs to just that set

jbw976 commented:
I realllllly like the idea of having Crossplane manage Crossplane's organization and metadata.  Super cool initiative here @hasheddan!! 💪 🙇‍♂️

hasheddan commented:
@jbw976 looks like branch protection is working correctly as I do not have permission to merge this :slightly_smiling_face: I believe I have addressed all comments so feel free to merge when ready!
